agent/taskresource/credentialspec/credentialspec_linux.go

```diff
@@ -0,0 +1,438 @@
+		}
+
+		credSpecValue := credSpecSplit[1]
+		if strings.HasPrefix(credSpecValue, "file://") {
+			wg.Add(1)
+			go cs.handleCredentialspecFile(credSpecStr, &wg, errorEvents)
+			continue
+		}
+
+		parsedARN, err := arn.Parse(credSpecValue)
+		if err != nil {
+			cs.setTerminalReason(err.Error())
+			return err
+		}
+		parsedARNService := parsedARN.Service
+		switch parsedARNService {
+		case "s3":
+			wg.Add(1)
+			go cs.handleS3CredentialspecFile(credSpecStr, credSpecValue, iamCredentials, &wg, errorEvents)
+		case "ssm":
+			wg.Add(1)
+			go cs.handleSSMCredentialspecFile(credSpecStr, credSpecValue, iamCredentials, &wg, errorEvents)
+		default:
+			err = errors.New("unsupported credentialspec ARN, only s3/ssm ARNs are valid")
+			cs.setTerminalReason(err.Error())
+			return err
+		}
+	}
+	wg.Wait()
+	close(errorEvents)
+	if len(errorEvents) > 0 {
+		var terminalReasons []string
+		for err := range errorEvents {
+			terminalReasons = append(terminalReasons, err.Error())
+		}
+
+		errorString := strings.Join(terminalReasons, ";")
+		cs.setTerminalReason(errorString)
+		return errors.New(errorString)
+	}
+
+	seelog.Infof("credentials fetcher daemon request: %v", cs.credentialsFetcherRequest)
+
+	// Check if skip credential fetcher invocation check override is present
+	skipSkipCredentialsFetcherInvocationCheck := utils.ParseBool(os.Getenv(envSkipCredentialsFetcherInvocation), false)
+	if skipSkipCredentialsFetcherInvocationCheck {
+		seelog.Info("Skipping credential fetcher invocation based on environment override")
+		testKrbFilePath := "/tmp/tgt"
+		os.Create(testKrbFilePath)
+		// assign temporary variable for test
+		cs.leaseid = "12345"
+		for k := range cs.ServiceAccountInfoMap {
+			cs.CredSpecMap[k] = testKrbFilePath
+		}
+		return nil
+	}
+
+	err := cs.handleKerberosTicketCreation()
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (cs *CredentialSpecResource) handleKerberosTicketCreation() error {
+	// Create kerberos tickets for the gMSA service accounts on the host location /var/credentials-fetcher/krbdir
+	if len(cs.credentialsFetcherRequest) > 0 {
+		//set up server connection to communicate with credentials fetcher daemon
+		conn, err := credentialsfetcherclient.GetGrpcClientConnection()
+		seelog.Infof("grpc connection: %v", conn)
+		if err != nil {
+			seelog.Errorf("failed to connect with credentials fetcher daemon: %s", err)
+			return err
+		}
+		// make the grpc call to add kerberos lease api to create kerberos tickets for the gmsa account
+		response, err := credentialsfetcherclient.NewCredentialsFetcherClient(conn, time.Minute).AddKerberosLease(context.Background(), cs.credentialsFetcherRequest)
+
+		if err != nil {
+			seelog.Errorf("failed to create kerberos tickets associated service account, error: %s", err)
+			cs.setTerminalReason(err.Error())
+			return err
+		}
+
+		cs.leaseid = response.LeaseID
+		seelog.Infof("credentials fetcher response leaseid: %v", cs.leaseid)
```

Comment:
please use consistent camel case for `leaseID` in log statements, the credspec resource struct and the response struct.

Reply:
made changes accordingly